.github/workflows/ci.yml

```diff
@@ -7,22 +7,40 @@ on:
     branches:
       - master
 jobs:
-  build:
+  test:
+    runs-on: ubuntu-20.04
+    name: Erlang/OTP ${{matrix.otp}}
     strategy:
       matrix:
-        platform: [ubuntu-18.04]
-        otp-version: [21.3.8.17-1, 22.3.4.7-1]
-    runs-on: ${{ matrix.platform }}
+        otp:
+          - '25'
+          - '24'
+          - '23'
+        rebar3:
+          - '3'
     steps:
-    - name: Prepare the environment
-      run: |
-        deb=esl-erlang_${{ matrix.otp-version }}~ubuntu~bionic_amd64.deb
-        wget https://packages.erlang-solutions.com/erlang/debian/pool/${deb}
-        sudo apt-get install --allow-downgrades --no-install-recommends ./${deb}
-        wget https://github.com/erlang/rebar3/releases/download/3.14.0/rebar3
-        chmod +x rebar3
-        sudo mv rebar3 /usr/local/bin/rebar3
-    - name: Checkout
-      uses: actions/checkout@v2
-    - name: Compile
-      run: make test
+      - uses: actions/checkout@v3
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: ${{matrix.otp}}
+          rebar3-version: ${{matrix.rebar3}}
+
+      - name: Rebar3 cache
+        uses: actions/cache@v1
+        env:
+          cache-name: rebar3
+        with:
+          path: ~/.cache/rebar3
+          key: ci-${{runner.os}}-${{env.cache-name}}-otp_${{matrix.otp}}-rebar_${{matrix.rebar3}}-${{hashFiles('rebar.lock')}}
+          restore-keys: |
+            ci-${{runner.os}}-${{env.cache-name}}-otp_${{matrix.otp}}-rebar_${{matrix.rebar3}}
+            ci-${{runner.os}}-${{env.cache-name}}-otp_${{matrix.otp}}
```

Comment:
This could be extended to cache the Dialyzer PLT too, I think.

Reply:
The intention is to cache the dialyzer PLT stored in `~/.cache/rebar3/*.plt`.

Reply:
Yes, but `~/.cache/rebar3` only contains the "base" PLT containing the OTP apps, and in that case it is unnecessary to key it on `rebar.lock`, you can just key it on the OTP version. (Unless I remember incorrectly how this caching thing works.)

There is another PLT stored under `_build/**/*_plt` which contains the PLT with the results from dependencies in it, and if you want to cache that, it would make sense to use `rebar.lock`. (In case you copied this from somewhere, I think this is a pretty common mistake when it comes to rebar3 and plt:s).

Reply:
Aha!
In [meck](https://github.com/eproxus/meck/blob/master/.github/workflows/erlang.yml#L35), everything under `_build` is cached, and then the ci starts by doing `make clean`. I only got the caching to work for directories, not for file name patterns. Do you think this is a good solution?

Reply:
Caching the entire `_build` directory is a bit smelly IMHO. The [cache action](https://github.com/actions/cache) says that `key` can be
> A list of files, directories, and wildcard patterns to cache and restore. See [@actions/glob](https://github.com/actions/toolkit/tree/main/packages/glob) for supported patterns.

Reply:
But if you can't get it to work, leave it. I doubt you'll gain more than like 5-10 seconds or so per build. :)

Reply:
My problem was with `path`.

Reply:
It would be nice to fix it so that it can be copied to other repositories later.

Reply:
https://github.com/klarna-incubator/system_monitor/pull/24